hack/deployment-diagnostics.sh

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+set -euo pipefail
+
+if [[ -z "${EV2:-}" ]]; then
+   # this script only executes in EV2
+   # executing this in other environments with less restricted access to CD logs
+   # can lead to leaking sensitive information
+   exit 0
+fi
+
+HELM_RELEASE_NAME="$1"
+NAMESPACE="$2"
+
+cat <<'EOF'
+
+
+  _          _                      _                       _        __
+ | |__   ___| |_ __ ___    _ __ ___| | ___  __ _ ___  ___  (_)_ __  / _| ___
+ | '_ \ / _ \ | '_ ` _ \  | '__/ _ \ |/ _ \/ _` / __|/ _ \ | | '_ \| |_ / _ \
+ | | | |  __/ | | | | | | | | |  __/ |  __/ (_| \__ \  __/ | | | | |  _| (_) |
+ |_| |_|\___|_|_| |_| |_| |_|  \___|_|\___|\__,_|___/\___| |_|_| |_|_|  \___/
+
+
+EOF
+
+# get release information as json
+RELEASE_INFO="$(helm status "$HELM_RELEASE_NAME" -n "${NAMESPACE}" -o json)"
+if [ -z "$RELEASE_INFO" ]; then
+  echo "Failed to retrieve Helm release information for release ${HELM_RELEASE_NAME} in namespace ${NAMESPACE}"
+  exit 1
+fi
+
+HELM_DEPLOYMENT_STATUS="$(jq --raw-output .info.status <<<"${RELEASE_INFO}")"
+HELM_DEPLOYMENT_DESCRIPTION="$(jq --raw-output .info.description <<<"${RELEASE_INFO}")"
+VALUES="$(jq --raw-output .config <<<"${RELEASE_INFO}")"
+echo ""
+echo "Status: ${HELM_DEPLOYMENT_STATUS}"
+echo "Description: ${HELM_DEPLOYMENT_DESCRIPTION}"
+echo "Release values:"
+echo "${VALUES}" | jq '.'
+
+cat <<'EOF'
+
+
+  _    ___            _ _                             _   _
+ | | _( _ ) ___    __| (_) __ _  __ _ _ __   ___  ___| |_(_) ___ ___
+ | |/ / _ \/ __|  / _` | |/ _` |/ _` | '_ \ / _ \/ __| __| |/ __/ __|
+ |   < (_) \__ \ | (_| | | (_| | (_| | | | | (_) \__ \ |_| | (__\__ \
+ |_|\_\___/|___/  \__,_|_|\__,_|\__, |_| |_|\___/|___/\__|_|\___|___/
+                                |___/
+
+
+EOF
+
+echo -e "\n--- Pods ---"
+kubectl get pods -n "$NAMESPACE" || echo "Could not list pods"
+
+DEPLOYMENTS=$(kubectl get deployments -n "$NAMESPACE" -o jsonpath='{.items[*].metadata.name}' 2>/dev/null || echo "")
+for DEPLOY in $DEPLOYMENTS; do
+    echo -e "\n--- Describe Deployment: $DEPLOY ---"
+    kubectl describe deployment "$DEPLOY" -n "$NAMESPACE"
+done
+
+PODS=$(kubectl get pods -n "$NAMESPACE" -o jsonpath='{.items[*].metadata.name}' 2>/dev/null || echo "")
+for POD in $PODS; do
+    echo -e "\n>>> Logs from pod: $POD"
```

Comment:
Is it possible to have output artifacts from Ev2? Might be nicer to have this as a downloadable file - all logs for all pods in the namespace could quickly balloon to megabytes of text and be annoying to operate over in a web view.

Reply:
as discussed in chat, it makes a lot of sense to craft links to Kusto instead of showing logs here.